static/panes/cfg-view.ts

```diff
@@ -22,427 +22,323 @@
-                    edge.physics = false;
-                } else {
-                    edge.physics = true;
-                    const diff = nodeB.level - nodeA.level;
-                    edge.length = diff * (200 - 5 * Math.min(5, diff));
+            if (keys.length > 0) {
+                if (selectedFunction === '' || !(selectedFunction !== null && selectedFunction in cfg)) {
+                    selectedFunction = keys[0];
                 }
+                this.functionSelector.setValue(selectedFunction, true);
+                this.state.selectedFunction = selectedFunction;
             } else {
-                edge.physics = false;
+                // this.state.selectedFunction won't change, next time the compilation results aren't errors or empty
+                // the selected function will still be the same
+                selectedFunction = null;
             }
+            this.selectFunction(selectedFunction);
+        } else {
+            // this case can be fallen into with a blank input file
+            this.selectFunction(null);
         }
     }
 
-    showCfgResults(data: vis.Data) {
-        this.assignLevels(data);
-        this.cfgVisualiser.setData(data);
-        /* FIXME: This does not work.
-         * It's here because I suspected that not having content in the constructor was
-         * breaking the move, but it does not seem like it
-         */
-        if (this.needsMove) {
-            this.cfgVisualiser.moveTo({
-                position: this.savedPos,
-                animation: false,
-                scale: this.savedScale,
-            });
-            this.needsMove = false;
+    async createBasicBlocks(fn: CfgDescriptor) {
+        for (const node of fn.nodes) {
+            const div = document.createElement('div');
+            div.classList.add('block');
+            div.innerHTML = await monaco.editor.colorize(node.label, 'asm', MonacoConfig.extendConfig({}));
+            if (node.id in this.bbMap) {
+                throw Error("Duplicate basic block node id's found while drawing cfg");
+            }
+            this.bbMap[node.id] = div;
+            this.blockContainer.appendChild(div);
+        }
+        for (const node of fn.nodes) {
+            const elem = $(this.bbMap[node.id]);
+            void this.bbMap[node.id].offsetHeight;
+            (node as AnnotatedNodeDescriptor).width = elem.outerWidth() as number;
+            (node as AnnotatedNodeDescriptor).height = elem.outerHeight() as number;
         }
     }
 
-    override onCompilerClose(compilerId: number) {
-        if (this.compilerInfo.compilerId === compilerId) {
-            // We can't immediately close as an outer loop somewhere in GoldenLayout is iterating over
-            // the hierarchy. We can't modify while it's being iterated over.
-            this.close();
-            _.defer(() => {
-                this.container.close();
-            }, this);
+    drawEdges() {
```

Comment:
Even though I read some of the algos you linked and the PR discuission, I'm not going to pretend I understand this code and trust you hahaha

Reply:
I am going to add some comments here because I will probably forget how everything works otherwise :)